tests/test_pyp2conf.py

```diff
@@ -271,3 +270,24 @@ def test_extras_detected_correctly():
     }
     pkg = PypiPackage("_", version="0", package_metadata=fake_pkg_data)
     assert pkg.extras() == ["dev", "docs", "lint", "test"]
+
+
+@pytest.mark.parametrize(
+    ("wheel_name", "archful"), [
+        ("sampleproject-3.0.0-py3-none-any.whl", False),
+        ("numpy-1.26.0-cp39-cp39-win_amd64.whl", True),
+        ("numpy-1.26.0-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.whl", True),
+        ("numpy-1.26.0-cp39-cp39-musllinux_1_1_x86_64.whl", True),
+    ]
+)
+def test_archfulness_is_detected(wheel_name, archful):
```

Comment:
All tested packages have exactly one entry in urls. A situation with multiple wheels is untested.